src/views/shared/client/utils.tsx

```diff
@@ -86,18 +87,36 @@ export function ReplaceNewlines(text: string): string {
     .join("");
 }
 
-export function LoadCardImageIntoElement(
+export async function LoadCardImageIntoElement(
   cardId: string,
   dataDetails: DataDetailsResponse,
   element: HTMLElement
-): void {
+): Promise<void> {
   if (element.getAttribute("data-loaded")) {
     return;
   }
   element.setAttribute("data-loaded", "true");
+  element.style.backgroundImage = "url(/Images/CardBack.jpg)";
```

Comment:
This immediately loads the card back image, which I don't think is what you want.

Reply:
The fact that all of our tests passed and the feature didn't actually work is a strong indication that we need a more specific test. Writing a wider scope test around LoadCardImageIntoelement will be non trivial, and in the current form, will probably require you to use puppeteer to create a page with some elements and stuff, and you'll have to intercept an image request to control how long the load takes.

It's going to be kind of a doozy but if you don't introduce the test in this branch it'll be putting a lot of trust on future-you to come back and do it later.

Reply:
Talked about this, I'll add an integration test for it and have you review it.